org-eclipse-che-jdt-ls-extension/jdt.ls.extension.api/src/main/java/org/eclipse/che/jdt/ls/extension/api/dto/CheWorkspaceEdit.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.api.dto;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.lsp4j.ResourceChange;
+import org.eclipse.lsp4j.TextDocumentEdit;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.xtext.xbase.lib.Pure;
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+/**
+ * It is temporary replacement of {@link org.eclipse.lsp4j.WorkspaceEdit} for transporting changes
+ * of the refactoring operation.
+ *
+ * <p>A workspace edit represents changes to many resources managed in the workspace. The edit
+ * should either provide `changes` or `documentChanges`. If documentChanges are present they are
+ * preferred over `changes` if the client can handle versioned document edits.
+ */
+@SuppressWarnings("all")
+public class CheWorkspaceEdit {
```

Comment:
As mentioned in the companion PR, we should add a "kind" when creating resources.

Reply:
Created a new class `CheResourceChange` which is extended `ResourceChange` and added a kind of the changed resource. Also reworked `CheWorkspaceEdit`. 
@tsmaeder please take a look.